src/python_bindings/py_algorithm.h

```diff
@@ -47,6 +87,77 @@ class PyAlgorithmBase {
     void LoadData();
 
     pybind11::int_ Execute(pybind11::kwargs const& kwargs);
+    std::unordered_map<std::string, std::string> GetOpts() {
+        std::unordered_map<std::string_view, boost::any> opt_ = algorithm_->GetOptValues();
+        std::unordered_map<std::string, std::string> res;
+        // for (auto& i:converters){
+        //     std::cout<<i.first<<" "<<'\n';
+        // }
+        for (auto& i : opt_) {
+            if (i.first == "table" || i.first == "input_format") {
+                continue;
+            }
+            if (i.first == "lhs_indices" || i.first == "rhs_indices") {
+                auto opt_value = (boost::any_cast<config::IndicesType>(i.second));
+                if (opt_value.size() == 0) {
+                    res[std::string(i.first)] = "[]";
+                    continue;
+                }
+                std::string str_value = "[";
+                for (int i = 0; i < opt_value.size() - 1; i++) {
+                    str_value += std::to_string(opt_value.at(i)) + ", ";
+                }
+                str_value += std::to_string(opt_value.back()) + "]";
+                res[std::string(i.first)] = str_value;
+                continue;
+            }
+
+            if (i.first == "metric") {
+                std::string str_val;
+                auto opt_val = boost::any_cast<algos::metric::Metric>(i.second);
+                switch (opt_val) {
```

Comment:
Use the `_to_string()` method of `BETTER_ENUM`
[This](http://aantron.github.io/better-enums/ApiReference.html) is the library used

Reply:
And move it into a separate function for enums (see `EnumConvPair` in `py_to_any.cpp`)